core/processor/src/main/java/io/quarkus/annotation/processor/documentation/config/scanner/ConfigMappingListener.java

```diff
@@ -152,8 +152,15 @@ public void onEnclosedMethod(DiscoveryRootElement discoveryRootElement, TypeElem
 
         AnnotationMirror configDocDefaultAnnotation = methodAnnotations.get(Types.ANNOTATION_CONFIG_DOC_DEFAULT);
         if (configDocDefaultAnnotation != null) {
-            builder.defaultValueForDoc(
-                    configDocDefaultAnnotation.getElementValues().values().iterator().next().getValue().toString());
+            String defaultValue = (String) utils.element().getAnnotationValues(configDocDefaultAnnotation).get("value");
+            if (defaultValue != null) {
```

Comment:
You don't need this as `value` is mandatory in the annotation, that's why we didn't test this `null` case.